README.md

```diff
@@ -10,126 +10,121 @@
-semver rules. This applies to the root [`./alias.go`](./alias.go) file as well.
+Where a full `fn` looks like
+`func(context.Context, event.Event) (*event.Event, transport.Result)`
 
-Even though `pkg/cloudevents` is v1.0.0, there could still be minor bugs and
-performance issues. We will continue to track and fix these issues as they come
-up. Please file a pull request or issue if you experience problems.
+For protocols that do not support responses, `StartReceiver` will throw an error
+when attempting to set a receiver fn with that capability.
 
-The API that exists under [`pkg/bindings`](./pkg/bindings) is a new API that
-will become SDK v2.x, and will replace `pkg/cloudevents`. This area is still
-under heavy development and will not be following the same semver rules as
-`pkg/cloudevents`. If a release is required to ship changes to `pkg/bindings`, a
-bug fix release will be issued (x.y.z+1).
+For protocols that do not support responses from send (Requester interface),
+`Client.Request` will throw an error.
 
-We will target ~2 months of development to release v2 of this SDK with an end
-date of March 27, 2020. You can read more about the plan for SDK v2 in the
-[SDK v2 planning doc](./docs/SDK_v2.md).
+**v2.0.0-preview1:**
 
-This SDK current supports the following versions of CloudEvents:
+In _preview1_ we are focusing on the new interfaces found in pkg/transport (will
+be renamed to protocol):
 
-- v1.0
-- v0.3
-- v0.2
-- v0.1
+- Sender, Send an event.
+- Requester, Send an event and expect a response.
+- Receiver, Receive an event.
+- Responder, Receive an event and respond.
 
 ## Working with CloudEvents
 
-Package [cloudevents](./pkg/cloudevents) provides primitives to work with
+Package [binding](./pkg/binding) provides primitives to bind an
+[event](./pkg/event) onto a [protocol](./pkg/transport) following the
 CloudEvents specification: https://github.com/cloudevents/spec.
 
+The SDK is written in such a way too allow as much control to the integrator as
+needed.
+
 Import this repo to get the `cloudevents` package:
 
 ```go
 import "github.com/cloudevents/sdk-go"
 ```
 
-Receiving a cloudevents.Event via the HTTP Transport:
+If all that is required is to convert a CloudEvent into JSON, use `event.Event`:
 
 ```go
-func Receive(event cloudevents.Event) {
-	// do something with event.Context and event.Data (via event.DataAs(foo)
-}
+event :=  cloudevents.NewEvent()
+event.SetSource("example/uri")
+event.SetType("example.type")
+event.SetData(cloudevents.ApplicationJSON, map[string]string{"hello": "world"})
 
-func main() {
-	c, err := cloudevents.NewDefaultClient()
-	if err != nil {
-		log.Fatalf("failed to create client, %v", err)
-	}
-	log.Fatal(c.StartReceiver(context.Background(), Receive));
-}
+bytes, err := json.Marshal(event)
 ```
 
-Creating a minimal CloudEvent in version 1.0:
+The SDK can help bind this to a specific protocol, in this case HTTP Request:
```

Comment:
Before this sentence, I would add just a quick explanation of what is `binding.Message`, something like:

> binding.Message is an abstraction to read a protocol specific message as a CloudEvent, without fully parsing it to `cloudevents.Event`, in order to allow optimized message passing between different protocols.